### PR TITLE
WIP: Port common makefile layout, add target and fixes

### PR DESCRIPTION
**Pull request by thebigcorporation**
- port common Makefile and base image layout to match other containers
 - fix errant latest tag: if no upstream is configured, a :latest tag is still applied this happens because the error message does not go to stdout, and the current code assumes there is no diff. fix by redirecting stderr to stdout
 - add a vanilla python target, and make sure that "python" maps to a valid commond using update-alternatives
 - todo: fix annopred test script

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,6 +4,7 @@ FROM $BASE_IMAGE
 
 RUN apt -y update -qq && apt -y upgrade && \
     DEBIAN_FRONTEND=noninteractive apt -y install \
-	apt-utils \
-    python3 \
-    python3-venv
+	curl ca-certificates \
+    	python3 python3-venv && \
+	update-alternatives --install /usr/bin/python python \
```
Comment (kscott-1) [inline Dockerfile]: This is now a good change now that any legacy python2 apps will be found elsewhere. Not that we should have any of those....

Format needs a quick fix it seems

File: Dockerfile.python
Comment (kscott-1) [inline Dockerfile.python:1]: Does this exist just for the test.sh feature?

File: Dockerfile.annopred
Comment (kscott-1) [inline Dockerfile.annopred:1]: This should hard code ubuntu 18.04 as base

File: src/test/annopred.sh
Comment (kscott-1) [inline src/test/annopred.sh:1]: I'll fix this, the target is just the annopred script

**Review by kscott-1 — COMMENTED** [review]
Another TODO: Think about consistent entrypoints across the python targets

**Comment by kscott-1** [comment]
`make docker_test` issue:

Testing Docker container: hihg-um/annopred:makefile_sync-0-g91cfa60-dirty
/bin/bash: /test.sh: Permission denied
Testing Docker container: hihg-um/python:makefile_sync-0-g91cfa60-dirty
/bin/bash: /test.sh: Permission denied
Testing Docker container: hihg-um/python-scripting:makefile_sync-0-g91cfa60-dirty
/bin/bash: /test.sh: Permission denied
make: *** [Makefile:86: docker_test] Error 126

Solution: add o+x to test.sh scripts or run the test container as root. test.sh will always show up as 750 root/root as configured right now.

**Comment by thebigcorporation** [comment]
The test thing is odd, since permissions in the repo are go+rx, so if this is an issue on clone, it should show up in every repo, and fix would be to explicitly set read-exec after copying, but might take some looking into.

File: src/test/annopred.sh
Comment (thebigcorporation) [reply src/test/annopred.sh:1]: cool. I ran out of steam past midnight, but it was a good run.

**Comment by kscott-1** [comment]
Yeah, not sure what happened with the perms, after fetch and pull, the test scripts are all 750.

File: Dockerfile.annopred
Comment (thebigcorporation) [reply Dockerfile.annopred:1]: I somehow dropped that in a merge - that was the intention, and why it's got its own update; upgrade apt section. Sorry.

**Comment by kscott-1** [comment]
Still TODO:

- Determine what fix to use for docker_test (I tested without the user mappings & had no issues)
- Entrypoint for each container

**Pull request closed without merging**